### PR TITLE
Terraform: Add support for using pre-existing private connectivity configuration

### PR DESCRIPTION
**Pull request by manitgupta**
1. Adds ability to specify a private connectivity configuration for the end-to-end infrastructure setup Terraform template. Private connectivity requires sensitive permissions hence the users might want to set it up manually instead of relying on automation.
2. Adds ability to configure `spanner_project_id` for cross-project writes.

**Note**: I had spent considerable time setting up MySQL 8.4 VPC setup by self-discovering how to configure a MySQL self-hosted database to work with Datastream. While I made numerous changes as per the official docs of MySQL, I am still running into errors. Current Datastream instructions are for MySQL 5.7 - https://cloud.google.com/datastream/docs/configure-self-managed-mysql. I will ask Datastream to provide documentation on how to configure MySQL 8, in the meantime switching back to MySQL 5.7.

### CODE REVIEW

File: v2/datastream-to-spanner/terraform/samples/mysql-end-to-end/main.tf [hunk 1]
```diff
@@ -39,14 +39,24 @@ resource "google_datastream_connection_profile" "source_mysql" {
...
+  # resource creation.
   dynamic "private_connectivity" {
     for_each = google_datastream_private_connection.datastream_private_connection.*.id
     content {
       private_connection = private_connectivity.value
     }
   }
 
+  # If an existing private connectivity configuration is provided, use that.
+  # If nothing is specified on private connectivity, IP whitelisting is
```
Comment (bharadwaj-aditya) [inline v2/datastream-to-spanner/terraform/samples/mysql-end-to-end/main.tf:51]: allowlisting.

**Review by bharadwaj-aditya — CHANGES_REQUESTED** [review]
wording change. Rest look fine.

File: v2/datastream-to-spanner/terraform/samples/mysql-end-to-end/main.tf (hunk 1 above)
Comment (manitgupta) [reply v2/datastream-to-spanner/terraform/samples/mysql-end-to-end/main.tf:51]: Done

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Pull request merged**